ui/src/app/components/runs/runs.component.ts

```diff
@@ -62,18 +69,52 @@ export class RunsComponent implements OnDestroy, AfterViewInit {
   }
 
   onClarityDgRefresh(state: ClrDatagridStateInterface) {
-    let sort: SortModel = state.sort ? new SortModel(<string>state.sort.by, state.sort.reverse ? -1 : 1) : undefined;
+    this.sort = state.sort ? new SortModel(<string>state.sort.by, state.sort.reverse ? -1 : 1) : undefined;
+    this.pageFrom = state.page.from < 0 ? 0 : state.page.from;
+    this.pageSize = state.page.size;
+
+    let filters:{[prop:string]: any} = {};
+    if (state.filters) {
+      for (let filter of state.filters) {
+        let {property, value} = <{property: string, value: any}>filter;
+        filters[property] = value;
+      }
+    }
+    this.filters = filters;
+
+    this.refresh();
+  }
 
-    let pageFrom = state.page.from < 0 ? 0 : state.page.from;
-    let pageSize = state.page.size;
+  refresh() {
+    let filters: FiltersModel = this.createFiltersModel(this.filters);
 
     let searchRequestModel: DagRunsSearchRequestModel = {
-      from: pageFrom,
-      size: pageSize,
-      sort: sort
+      from: this.pageFrom,
+      size: this.pageSize,
+      sort: this.sort,
+      filters: filters
     };
 
     this.store.dispatch(new GetDagRuns(searchRequestModel));
   }
 
+  createFiltersModel(filters: {[prop:string]: any}): FiltersModel {
+    let byWorkflowNameOption = this.filters[dagRunColumns.WORKFLOW_NAME];
```

Comment:
You don't need `this.`

Reply:
Can you mark the method private?

Reply:
Also, consider using `const` instead of `let`

Reply:
Fixed